app/api/ipfs/generate-jwt/route.ts

```diff
@@ -0,0 +1,29 @@
+import getIsPro from "@/lib/getIsPro";
+import generatePinataJWT from "@/lib/ipfs/generatePinataJWT";
+import { NextRequest } from "next/server"
+import { isAddress } from "viem"
+
+export async function GET(req: NextRequest) {
+  const address = req.nextUrl.searchParams.get('address')
+  if (!isAddress(address)) {
+    return Response.json({ message: "address required" }, { status: 422 });
+  }
+
+  try {
+    const { isPro } = await getIsPro(address)
+    if (!isPro) {
+      return Response.json({ message: "not on pro plan" }, { status: 403 });
+    }
```

Comment:
if we are now validating `domain` in middleware, do we need to verify `isPro` here?
- we could just verify in `client` when it's needed right?